ocaml/testsuite/tests/typing-modes/mutable.ml

```diff
@@ -81,14 +81,13 @@ Line 1, characters 26-29:
 Error: This value is nonportable but expected to be portable.
 |}]
 
-(* For monadic axes, mutable defaults to mutable(min). So currently we can't
-   write a [contended] value to a mutable field. *)
-let foo (r @ uncontended) (s @ contended) = r.s <- s
+(* This attribute doesn't disable implied modalities on monadic axes. For
```

Comment:
This PR doesn't affect the contention axis, because the legacy modality is `uncontended` which is identity modality.

The test here wants to show that there is a `shared` modality. Several ways to show this:
- During construction, that means we allow field to be `shared` when constructing a `unique` record.
- During projection, that means field is `shared` even if the record is `unique`.
- During mutation, as shown in this example.

The first two relies on the record being unique, which is tricky as you said. The last one doesn't, and:
- If there is no `shared` modality, then the new value is required to be `unique`, then this test will fail.
- If there is `shared` modality, then the new value is required to be `shared`, then this test succeeds.

That being said, I'm still happy to add the first two as tests.

Reply:
Maybe I'm misunderstanding the test then? When I test both this and the same but with `@ contended`, I get the same behavior (on `main`). Yet it sounds like I should get different behavior.

Let me see if I understand:
* The mutability mode on monadic axes is `min`, right? So that would mean that any value written to a mutable field would have to be `unique` and `uncontended` -- regardless of the mode of the record itself.
* Except that if there is a modality `m`, we can write at mode `m(min)`. So the implied `shared` modality means we can write `shared`. But there is no implied `contended` modality.... so why can I write `contended`?

I looked at the "Modes and mutation" design doc, but it doesn't really cover modalities.

Reply:
Let's have a call.

Reply:
We discovered in the call two things:
1. I made a mistake here by testing with `string`, which mode-crosses contention.
2. It would be useful to have the typing rules around mutability written down, so it's easier to see how all these pieces fit together.

Reply:
(resolved offline) and I will add tests.

Reply:
The internal document is now updated with a more detailed descirption of what we think monadic axes should react with mutable. I found plain English better than typing rules.

I also briefly describe the current implementation of mutable vs. modes.